File: MultiLanguageSandbox/src/test/deps/swift/44.swift
func convertIntToBoolToInt(_ value: Int) -> Int {
    let boolValue = value != 0
    return boolValue ? 1 : 0
}


func testConvertIntToBoolToInt() {
    assert(convertIntToBoolToInt(5) == 1, "Conversion of 5 should return 1.")
    assert(convertIntToBoolToInt(0) == 0, "Conversion of 0 should return 0.")
    assert(convertIntToBoolToInt(-10) == 1, "Conversion of -10 should return 1.")
    assert(convertIntToBoolToInt(1) == 1, "Conversion of 1 should return 1.")
    assert(convertIntToBoolToInt(-1) == 1, "Conversion of -1 should return 1.")
    assert(convertIntToBoolToInt(100) == 1, "Conversion of 100 should return 1.")
    assert(convertIntToBoolToInt(-100) == 1, "Conversion of -100 should return 1.")
}

// Call the test function to verify the functionality
testConvertIntToBoolToInt()

File: MultiLanguageSandbox/src/test/deps/elixir/44.exs
defmodule TypeConversion do
def int_to_bool_to_int(value) when is_integer(value) do
    # Convert integer to boolean (non-zero becomes true, zero becomes false)
    bool_value = value != 0
    
    # Convert boolean back to integer (true becomes 1, false becomes 0)
    if bool_value, do: 1, else: 0
  end
end



ExUnit.start()
defmodule TypeConversionTest do
  use ExUnit.Case

  test "int_to_bool_to_int returns 0 for input 0" do
    assert TypeConversion.int_to_bool_to_int(0) == 0
  end

  test "int_to_bool_to_int returns 1 for positive integers" do
    assert TypeConversion.int_to_bool_to_int(1) == 1
    assert TypeConversion.int_to_bool_to_int(10) == 1
  end

  test "int_to_bool_to_int returns 1 for negative integers" do
    assert TypeConversion.int_to_bool_to_int(-1) == 1
    assert TypeConversion.int_to_bool_to_int(-10) == 1
  end
end